src/Components/ExtractLevelsMenu.jsx

```diff
@@ -0,0 +1,184 @@
+    switch (normalDirection) {
+      case 'x':
+        normal = new Vector3(1, 0, 0)
+        break
+      case 'y':
+        normal = new Vector3(0, 1, 0)
+        break
+      case 'z':
+        normal = new Vector3(0, 0, 1)
+        break
+      default:
+        normal = new Vector3(0, 1, 0)
+        break
+    }
+    console.log('in the function normal', normal)
+    if (!planeHash || planeHash !== normalDirection ) {
+      addHashParams(window.location, PLANE_PREFIX, {planeAxis: normalDirection})
+    }
+    return viewer.clipper.createFromNormalAndCoplanarPoint(normal, modelCenter)
+  }
+  useEffect(() => {
+    const planeHash = getHashParams(location, 'p')
+    if (planeHash && model && viewer) {
+      const modelCenter = getModelCenter(model)
+      const planeDirection = planeHash.split(':')[1]
+      let normal
+      switch (planeDirection) {
+        case 'x':
+          normal = new Vector3(1, 0, 0)
+          break
+        case 'y':
+          normal = new Vector3(0, 1, 0)
+          break
+        case 'z':
+          normal = new Vector3(0, 0, 1)
+          break
+        default:
+          normal = new Vector3(0, 1, 0)
+          break
+      }
+      console.log(' ------------------------- ')
+      console.log('the modelCenter from the useEffect', modelCenter)
+      console.log('plane direction', planeDirection)
+      console.log('normal', normal)
+      console.log(' ------------------------- ')
+      createPlane(planeDirection)
+      // viewer.clipper.createFromNormalAndCoplanarPoint(normal, modelCenter)
+    }
+  }, [model])
+
+  showExtractMenu = async () => {
+    const allStor = await extractHeight(model)
+    try {
+      if (floorplanMenuItems.length + 1 <= allStor.length) {
+        const planeoffset = 0.5
+        for (let i = 0; i < allStor.length; i++) {
+          floorplanMenuItems[i] = (
+            <MenuItem onClick={() =>
+              createFloorplanPlane(allStor[i], allStor[i + 1] - planeoffset)}
+            >  L{i} </MenuItem>)
+        }
+      }
+    } catch {
+      console.log('No levels found')
+    }
+  }
+
+  showExtractMenu()
+
+  return (
+    <div>
+      <TooltipIconButton
+        title={'Extract Levels'}
+        icon={<LevelsIcon/>}
+        onClick={handleClick}
+      />
+      <Menu
+        elevation={1}
+        id='basic-menu'
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+        anchorOrigin={{vertical: 'top', horizontal: 'center'}}
+        transformOrigin={{vertical: 'top', horizontal: 'center'}}
+        className={classes.root}
+        PaperProps={{
+          style: {
+            left: '300px',
+            transform: 'translateX(-40px) translateY(-40px)',
+          },
+        }}
+      >
+        <TooltipIconButton
+          title={'Toggle Plan View'}
+          icon={<PlanViewIcon/>}
+          onClick={planView}
+        />
+        {floorplanMenuItems}
+      </Menu>
+    </div>
+  )
+}
+
+const useStyles = makeStyles({
```

Comment:
this form of styles is being deprecated.  New way is just to use an "sx" property on the element that you want to be styled.

https://mui.com/system/getting-started/the-sx-prop/

Reply:
Can you elaborate? I'm new to styles.